### PR TITLE
Entity removal

### LINKED ISSUE #22: Deleting an entity from an existing model
When an entity definition is removed (e.g. from a .go file, or the .go file defining an entity is removed altogether), the user might want to delete the entity from the model. Currently, the generator doesn't recognize this situation because it's executed for each individual input file, combining all of them into the same model - therefore it doesn't distinguish between missing entities and the ones that haven't been processed yet.

There are a couple of ways to approach this, the simplest would probably be adding a new CLI argument/action, sth. like `--remove=EntityName`. Alternatively, we could change the way the generator is launched so that it has the complete list of files to include in the model.

### PR DESCRIPTION
**Pull request by vaind**
Closes #22 

Adds support for: 
* providing directory/Glob pattern to the generator
* removing missing entities (if a dir/pattern is given)

Keeps existing behaviour without change (when called with a single file) - entities are not removed from the model but built-up incrementally (when called on multiple fbs files)

### CODE REVIEW

File: internal/generator/generator.go [hunk 1]
```diff
@@ -125,35 +143,106 @@ func Process(options Options) error {
...
-	}
+	return pathForEach(options.InPath, func(filePath string) error {
+		if !options.CodeGenerator.IsSourceFile(filePath) {
+			return nil
+		}
 
-	if err = mergeBindingWithModelInfo(currentModel, storedModel); err != nil {
-		return fmt.Errorf("can't merge model information: %s", err)
-	}
+		// clear meta information from the previous creatBinding() call (when processing multiple files at once)
```
Comment (greenrobot-team) [inline internal/generator/generator.go:151]:
```suggestion
		// clear meta information from the previous createBinding() call (when processing multiple files at once)
```
If there is time fix me.

Comment (greenrobot-team) [inline internal/generator/generator.go:151]:
```suggestion
		// clear meta information from the previous createBinding() call (when processing multiple files at once)
```
If there is time fix me.

File: test/integration/typeful/cpp/objectbox-model.backup.json [hunk 2]
```diff
@@ -0,0 +1,217 @@
+{
```
Comment (greenrobot-team) [inline test/integration/typeful/cpp/objectbox-model.backup.json:1]: Why was this file committed? Am I missing something?

**Review by greenrobot-team — COMMENTED** [review]
Only typo and a question, otherwise 👍  Should it be documented in the command help more clearly that one can remove entities using a directory/glob path?

File: internal/generator/generator.go (hunk 1 above)
Comment (vaind) [reply internal/generator/generator.go]: fixed

File: test/integration/typeful/cpp/objectbox-model.backup.json (hunk 2 above)
Comment (vaind) [reply test/integration/typeful/cpp/objectbox-model.backup.json]: by mistake, removed

**Comment by vaind** [comment]
> Should it be documented in the command help more clearly that one can remove entities using a directory/glob path?

I've updated the docs a little:
```
 ./objectbox-generator --help
Usage:
  objectbox-generator [flags] {path}
      * to execute "clean" action (see below) on the path, removing previously generated code and missing entities,
      * and execute code generation on the path afterwards.

      The given {path} can be one of the following:
        * directory - a non-recursive clean and generation is performed on the given directory,
        * a glob path pattern (e.g. contains a "*") - performs clean and generation on the matching paths,
        * a Go-style path pattern (e.g. "./..." - a recursive match of the current dir) - performs clean and generation on the matching paths,


or
  objectbox-generator [flags] {model/file/path.fbs}
      to generate the binding code for a single file


or
  objectbox-generator [flags] clean {path}
      to remove the generated files instead of creating them - this removes *.obx.* and objectbox-model.h but keeps objectbox-model.json

or
  objectbox-generator FLATC [flatc arguments]
      to execute FlatBuffers flatc command line tool Any arguments after the FLATC keyword are passed through.

path:
  * a source file path or a valid path pattern (e.g. ./...)
  
Available flags:
  -c    generate plain C code
  -cpp
        generate C++ code
  -go
        generate Go code
  -help
        print this help
  -optional string
        C++ wrapper type to use for fields annotated "optional"; one of: std::optional, std::unique_ptr, std::shared_ptr
  -out string
        output path for generated source files
  -persist string
        path to the model information persistence file (JSON)
  -version
        print the generator version info
```

**Comment by greenrobot-team** [comment]
> I've updated the docs a little:

👍

**Pull request merged**